src/H5FDint.c

```diff
@@ -1555,7 +1562,7 @@ H5FD__write_selection_translate(H5FD_t *file, H5FD_mem_t type, hid_t dxpl_id, ui
             io_len = MIN(file_len[file_seq_i], mem_len[mem_seq_i]);
 
             /* Check if we're using vector I/O */
-            if (use_vector) {
+            if (!skip_vector_cb && use_vector) {
```

Comment:
Should use skip_vector_cb to determine the value of use_vector on line 1460, then there will be no need to keep checking it

Reply:
Fixed